src/node/internal/internal_fs_promises.ts

```diff
@@ -254,76 +260,94 @@ export class FileHandle extends EventEmitter {
+      }
 
-    const bytesWritten = fssync.writeSync(
-      this.#fd,
-      buffer,
-      offsetPositionOrOptions,
-      lengthOrEncoding,
-      position
-    );
-
-    return {
-      bytesWritten,
-      buffer,
-    };
+      const bytesWritten = fssync.writeSync(
+        this.#fd,
+        buffer,
+        offsetPositionOrOptions,
+        lengthOrEncoding,
+        position
+      );
+
+      return Promise.resolve({
+        bytesWritten,
+        buffer,
+      });
+    } catch (err) {
+      return Promise.reject(err as Error);
+    }
   }
 
-  async writev(
+  writev(
     buffers: NodeJS.ArrayBufferView[],
     position: Position = null
   ): Promise<{ bytesWritten: number; buffers: NodeJS.ArrayBufferView[] }> {
-    if (this.#fd === undefined) {
-      throw new ERR_EBADF({ syscall: 'stat' });
+    try {
+      if (this.#fd === undefined) {
+        throw new ERR_EBADF({ syscall: 'stat' });
+      }
+      const bytesWritten = fssync.writevSync(this.#fd, buffers, position);
+      return Promise.resolve({
+        bytesWritten,
+        buffers,
+      });
+    } catch (err) {
+      return Promise.reject(err as Error);
     }
-    const bytesWritten = fssync.writevSync(this.#fd, buffers, position);
-    return {
-      bytesWritten,
-      buffers,
-    };
   }
 
-  async writeFile(
+  writeFile(
     data: string | Buffer,
     options: ValidEncoding | WriteFileOptions = {}
   ): Promise<{ bytesWritten: number; buffer: Buffer }> {
-    if (this.#fd === undefined) {
-      throw new ERR_EBADF({ syscall: 'stat' });
+    try {
+      if (this.#fd === undefined) {
+        throw new ERR_EBADF({ syscall: 'stat' });
+      }
+      const bytesWritten = fssync.writeFileSync(this.#fd, data, options);
+      return Promise.resolve({
+        bytesWritten,
+        buffer: isArrayBufferView(data)
+          ? data
+          : Buffer.from(data, options as BufferEncoding),
+      });
+    } catch (err) {
+      return Promise.reject(err as Error);
     }
-    const bytesWritten = fssync.writeFileSync(this.#fd, data, options);
-    return {
-      bytesWritten,
-      buffer: isArrayBufferView(data)
-        ? data
-        : Buffer.from(data, options as BufferEncoding),
-    };
   }
 
-  async close(): Promise<void> {
-    this.#handle?.close();
-    this.#fd = undefined;
-    this.#handle = undefined;
-    (this as unknown as EventEmitter).emit('close');
+  close(): Promise<void> {
+    try {
+      this.#handle?.close();
+      this.#fd = undefined;
+      this.#handle = undefined;
+      (this as unknown as EventEmitter).emit('close');
+      return Promise.resolve();
+    } catch (err) {
+      return Promise.reject(err as Error);
+    }
   }
 
   async [Symbol.asyncDispose](): Promise<void> {
     await this.close();
   }
 
-  readableWebStream(_options: any = {}): void {
+  readableWebStream(_options: ReadableWebStreamOptions = {}): void {
+    // TODO(node-fs): Implement
```

Comment:
Can you be a little bit more specific

Reply:
As soon as this is landed i'm opening a PR that implements so it would be somewhat pointless :-)